Feat/gen 8 data alex whan

alex-whan opened the pull request:
I've added all of the data I could find (to the best of my ability) involving the new Pokemon introduced in the recently-released Crown Tundra expansion: Regieleki, Regidrago, Glastrier, Spectrier, Calyrex, and Calyrex's two forms (Ice Rider Calyrex & Shadow Rider Calyrex).

All of this data was collected manually from Bulbapedia. Please note that I have designated the "ids" of Ice Rider Calyrex & Shadow Rider Calyrex as "20000" in all applicable locations in the CSV files, as I was unable to find a reliable source for this "forme" data, and instead followed the placeholder structure of the previous Regieleki entry.

On that note: Regieleki was already present in the "pokemon.csv" file, albeit with incomplete information. I've updated that record with the new/full Crown Tundra data.

Lastly: I've incorporated all data in the appropriate places within all of the CSV files I could find that were relevant. Please let me know if I should pull this data out and submit it separately. I've ordered Regieleki through Calyrex by their Galar Pokedex IDs (#894 through #898), and have placed them in the appropriate order following Zarude at #893 wherever possible. For the "formes" of Ice Rider Calyrex & Shadow Rider Calyrex, I have labeled them with ID "20000" and placed them at the very end of their respective CSV files. Please let me know if I should submit this data differently.

Also note: I did not submit a second PR on the main PokeAPI repository as no "logic" was changed - just data entry.

Naramsim commented:
Thanks so much for this update @alex-whan! I need to merge some data into this repo before merging your PR. I'll try my best to do it this afternoon. 

If there will be any conflicts someone of us two will fix them 👍

Naramsim commented on pokedex/data/csv/pokemon.csv:
```diff
@@ -1047,4 +1052,5 @@ id,identifier,species_id,height,weight,base_experience,order,is_default
 10157,necrozma-ultra,800,75,2300,339,956,0
-20000,regieleki,20000,12,1450,,,1
+20000,calyrex-ice-rider,20000,24,8091,,,1
+20000,calyrex-shadow-rider,20000,24,536,,,1
```
We can't have two `20000`.

Naramsim commented on pokedex/data/csv/abilities.csv:
```diff
@@ -292,3 +292,8 @@ id,identifier,generation_id,is_main_series
 10058,shackle,5,0
 10059,decoy,5,0
 10060,shield,5,0
+20000,transistor,8,1
```
So yes, here you can use the IDs 262-266

Naramsim commented on pokedex/data/csv/pokemon_species.csv:
```diff
@@ -892,3 +892,10 @@ id,identifier,generation_id,evolves_from_species_id,evolution_chain_id,color_id,
+896,glastrier,8,,,9,,,3,,0,0,,,1,0,1,,,
+897,spectrier,8,,,1,,,3,,0,0,,,1,0,1,,,
+898,calyrex,8,,,5,,,3,,,0,,,1,1,1,,,
+20000,calyrex-ice-rider,8,,,,,,3,,,0,,,1,1,1,,,
```
let's not put the ice-rider and shadow-rider in this species file.

Naramsim commented:
@alex-whan, for the abilities' IDs, you can use the one used by Bulbapedia: https://bulbapedia.bulbagarden.net/wiki/Ability#List_of_Abilities

Naramsim commented on pokedex/data/csv/pokemon_species_flavor_text.csv:
```diff
@@ -188199,3 +188199,17 @@ Pokémon has raised since the child’s infancy."
+897,34,9,"As it dashes through the night, Spectrier absorbs the life-force of sleeping creatures. It craves silence and solitude."
+898,33,9,"Calyrex is a merciful Pokémon, capable of providing healing and blessings. It reigned over the Galar region in times of yore."
+898,34,9,"Calyrex is known in legend as a king that ruled over Galar in ancient times. It has the power to cause hearts to mend and plants to spring forth."
+20000,33,9,"According to lore, this Pokémon showed no mercy to those who got in its way, yet it would heal its opponents' wounds after battle."
```
Same here, let's stick with the standard pokemon, so no variants.

Naramsim replied:
please, get rid of the 2 lines.

Naramsim replied:
Here :)

alex-whan replied:
Fixed it! Thanks. Removed the two Shadow/Ice Rider Calyrex form descriptions fro m the species flavor text.

Naramsim commented on pokedex/data/csv/pokemon_species_names.csv:
```diff
@@ -8562,3 +8562,10 @@ pokemon_species_id,local_language_id,name,genus
+896,9,glastrier,Wild Horse Pokémon
+897,9,spectrier,Swift Horse Pokémon
+898,9,calyrex,King Pokémon
+20000,9,calyrex-ice-rider,High King Pokémon
```
Same here

Naramsim commented on pokedex/data/csv/pokemon_stats.csv:
```diff
@@ -6287,3 +6317,15 @@ pokemon_id,stat_id,base_stat,effort
+20000,3,80,0
+20000,4,165,3
+20000,5,100,0
+20000,6,150,0
```
Here it will be 20000 and 20001

Naramsim commented on pokedex/data/csv/pokemon_types.csv:
```diff
@@ -1605,3 +1611,7 @@ pokemon_id,type_id,slot
+20000,14,1
+20000,15,2
+20000,14,1
+20000,8,2
```
And also here 20000 and 20001

alex-whan commented:
> Thanks so much for this update @alex-whan! I need to merge some data into this repo before merging your PR. I'll try my best to do it this afternoon.
> 
> If there will be any conflicts someone of us two will fix them 👍

Hi @Naramsim - thanks so much for the reviews and feedback, and the quick turnaround! I've noted the changes that need to be made and I'll work on those right away and let you know once those are fixed!

alex-whan commented:
Hi @Naramsim, I've gone ahead and made the requested changes. Updated all IDs for Ice Rider/Shadow Rider Calyrex to 20000 and 20001, respectively, updated the new Crown Tundra abilities to match up with Bulbapedia's ability IDs, and removed Ice Rider/Shadow Rider Calyrex forms from all indicated species files. Please let me know if all looks good!

Naramsim commented on package-lock.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "lockfileVersion": 1
```
Let's remove the packages, both. You aren't adding any new JS module. Or, if you do, you could put in in our scripts folder.

alex-whan replied:
Done!

Naramsim commented on pokedex/data/csv/ability_flavor_text.csv:
```diff
@@ -17469,3 +17469,9 @@ les attaques super efficaces."
+20000,20,9,"When the Pokémon knocks out a target, it utters a chilling neigh, which boosts its Attack stat."
+20000,20,9,"When the Pokémon knocks out a target, it utters a terrifying neigh, which boosts its Sp. Atk stat."
+20000,20,9,"This Ability combines the effects of both Calyrex's Unnerve Ability and Glastrier's Chilling Neigh Ability."
+20000,20,9,"This Ability combines the effects of both Calyrex's Unnerve Ability and Spectrier's Grim Neigh Ability."
```
It's impossible that we have `20000` for every ability added. 

those 20000s should become 262-266.

alex-whan replied:
Hi @Naramsim - how would you like to handle the "As One" ability for Shadow/Ice Rider Calyrex formes? It's a weird situation - Bulbapedia has it listed as "266/267" since it's technically a combination of two different abilities, with two permutations: 266 (As One/Chilling Neigh) and 267 (As One/Grim Neigh). Please advise on how I should set that up - for the moment, I've added a "267" to the ability_flavor_text.csv.

Naramsim replied:
Hi @alex-whan, thanks! I would then duplicate the `As One` ability and add an identical 267 ID.

Naramsim replied:
In the `ability_names` and `abilities` CSV files.

alex-whan replied:
Hi @Naramsim, got it, thank you. I've created the duplicate listings for the As One ability for #267 and added them to the `ability_names` and `abilities` CSV files. Change is committed.

Naramsim commented on pokedex/data/csv/pokemon.csv:
```diff
@@ -1049,3 +1054,5 @@ id,identifier,species_id,height,weight,base_experience,order,is_default
 10155,necrozma-dusk,800,38,4600,306,954,0
 10156,necrozma-dawn,800,42,3500,306,955,0
 10157,necrozma-ultra,800,75,2300,339,956,0
+20000,calyrex-ice-rider,20000,24,8091,,,1
```
I expect `calyrex-ice-rider` to have `species_id` equal to `898`. Same for the `shadow-rider`

alex-whan replied:
@Naramsim Understood, thanks. I'll make the requested changes!

alex-whan replied:
Changed to the correct `species_id` of 898 for both `shadow-rider` and `ice-rider` forms of Calyrex.

alex-whan commented:
Hi @Naramsim, hope all is well. Just wanted to check in and see if there were any further edits I should make to this PR! I reviewed it again this morning, and the commits should cover all of the revisions so far. Thanks!

alex-whan commented:
@Naramsim I was out of town the last few days but I am back in action now if there are any edits needed on this PR! I think everything should be good, though.

Naramsim commented:
Thanks so much, Alex, and accept my apologies for all the late responses.

alex-whan commented:
@Naramsim no worries at all - thanks for merging it in! Happy to help and I'll look for an opportunity to help out again soon with PokeAPI however I can. Please let me know if there's anything code-wise in particular that still needs doing.
